internal/dataplane/translator/subtranslator/httproute.go

```diff
@@ -592,31 +591,31 @@ func generateHeaderModifierKongPlugin(modifier *gatewayapi.HTTPHeaderFilter, plu
 
 	// modifier.Set is converted to a pair composed of "replace" and "add"
 	if modifier.Set != nil {
-		setModifiers := make([]string, 0, len(modifier.Set))
+		setModifiers := make([]interface{}, 0, len(modifier.Set))
 		for _, s := range modifier.Set {
 			setModifiers = append(setModifiers, kongHeaderFormatter(s))
 		}
-		plugin.Config["replace"] = map[string][]string{
+		plugin.Config["replace"] = map[string]interface{}{
 			"headers": setModifiers,
 		}
-		plugin.Config["add"] = map[string][]string{
+		plugin.Config["add"] = map[string]interface{}{
 			"headers": setModifiers,
 		}
 	}
 
 	// modifier.Add is converted to "append"
 	if modifier.Add != nil {
-		appendModifiers := make([]string, 0, len(modifier.Add))
+		appendModifiers := make([]interface{}, 0, len(modifier.Add))
 		for _, a := range modifier.Add {
 			appendModifiers = append(appendModifiers, kongHeaderFormatter(a))
 		}
-		plugin.Config["append"] = map[string][]string{
+		plugin.Config["append"] = map[string]interface{}{
 			"headers": appendModifiers,
 		}
 	}
 
 	if modifier.Remove != nil {
-		plugin.Config["remove"] = map[string][]string{
+		plugin.Config["remove"] = map[string]interface{}{
```

Comment:
Are we able to use a generic type that supports multiple options instead of straight `interface{}`? I expect there's a hidden limitation on what can actually go into the config here, and having the actual options spelled out should make that clearer in the future.

Reply:
54e8fa9 should cover this.